File: _2020/go/day1/day1.go
package day1

import (
	"fmt"
	"strconv"
	"strings"

	"advent/utils"
)

func parseInput() []int {
	data := utils.LoadPuzzleInput("day1")
	input_strings := strings.Fields(data)
	integers := make([]int, len(input_strings))
	for ind, str := range input_strings {
		i, _ := strconv.Atoi(str)
		integers[ind] = i
	}
	return integers
}

func Part1() string {
	integers := parseInput()
	value := 0
	for _, a := range integers {
		for _, b := range integers {
			if a+b == 2020 {
				value = a * b
				break
			}
		}
	}
	return fmt.Sprintf("%v", value)
}

func Part2() string {
	integers := parseInput()
	value := 0
	for _, a := range integers {
		for _, b := range integers {
			for _, c := range integers {
				if a+b+c == 2020 {
					value = a * b * c
					break
				}
			}
		}
	}
	return fmt.Sprintf("%v", value)
}

func All() {
	utils.Profile(Part1)
	utils.Profile(Part2)
}

File: python/puzzles/y2020/day1.py
import itertools

from utils import load_puzzle_input, profile


def process_input(raw: str):
    return list(map(int, raw.split("\n")))


@profile
def part1(raw: str):
    for a, b in itertools.combinations(process_input(raw), 2):
        if a + b == 2020:
            return a * b


@profile
def part2(raw: str):
    for a, b, c in itertools.combinations(process_input(raw), 3):
        if a + b + c == 2020:
            return a * b * c


if __name__ == "__main__":
    raw = load_puzzle_input("2020/day1")
    part1(raw)
    part2(raw)
